pyrometheus/__init__.py

```diff
@@ -178,8 +179,93 @@ def equilibrium_constants_expr(sol: ct.Solution, react: ct.Reaction, gibbs_rt):
 
 # {{{ Rate coefficients
 
-def arrhenius_expr(rate_coeff: ct.Arrhenius):
-    raise NotImplementedError()
+def rate_coefficient_expr(rate_coeff: ct.Arrhenius, t):
+    """This function returns rate-coefficient expressions"""
+    # Rate parameters
+    a = rate_coeff.pre_exponential_factor
+    b = rate_coeff.temperature_exponent
+    t_a = rate_coeff.activation_energy/ct.gas_constant
+    if b == 0.0 and t_a == 0.0:
+        # Constant rate
+        return a
+    elif b != 0.0 and t_a == 0.0:
+        # Weakly temperature-dependent rate
+        return a * t**b
+    elif b == 0.0 and a != 0.0 and t_a != 0.0:
+        # Classic Arrhenius rate
+        return p.Variable("exp")(np.log(a)-t_a/t)
+    else:
+        # Modified Arrhenius
+        return p.Variable("exp")(np.log(a)+b*p.Variable("log")(t)-t_a/t)
+
+
+def third_body_efficiencies_expr(sol: ct.Solution, react: ct.Reaction, c):
+    """This function returns an expression for the third-body rate factor"""
+    efficiencies = [react.efficiencies[sp] for sp in react.efficiencies]
+    indices_nondef = [sol.species_index(sp) for sp in react.efficiencies]
+    indices_default = np.delete(np.linspace(0, sol.n_species-1, sol.n_species,
+                                            dtype=int), indices_nondef)
+    sum_nondef = sum(eff_i * c[index_i] for eff_i, index_i
+                     in zip(efficiencies, indices_nondef))
+    sum_default = react.default_efficiency * sum(c[i] for i in indices_default)
+    return sum_nondef + sum_default
+
+
+def troe_falloff_expr(react: ct.Reaction, t):
+    """This function returns Troe-falloff expression"""
+    troe_params = react.falloff.parameters
+    troe_1 = (1.0-troe_params[0])*p.Variable("exp")(-t/troe_params[1])
+    troe_2 = troe_params[0]*p.Variable("exp")(-t/troe_params[2])
+    if troe_params[3] > 1.0e-16:
+        troe_3 = p.Variable("exp")(-troe_params[3]/t)
+        return troe_1 + troe_2 + troe_3
+    else:
+        return troe_1 + troe_2
+
+# }}}
+
+
+# {{{ Rates of progress
+
+def rate_of_progress_expr(sol: ct.Solution, react: ct.Reaction, c, k_fwd, k_eq):
+    """This function returns an expression for the reaction rate of progress"""
+    indices_reac = [sol.species_index(sp) for sp in react.reactants]
+    indices_prod = [sol.species_index(sp) for sp in react.products]
+
+    if len(react.orders) > 0:
```

Comment:
```suggestion
    if react.orders:
```
https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false